File: java/datastructures/Queue.java
// Queue implementation in Java
// FIFO data structure with enqueue, dequeue, peek, isEmpty, size, clear, and toArray methods

import java.util.*;

public class Queue<T> {
    private List<T> items;

    public Queue() {
        this.items = new ArrayList<>();
    }

    public void enqueue(T element) {
        items.add(element);
    }

    public T dequeue() {
        if (isEmpty()) {
            return null;
        }
        return items.remove(0);
    }

    public T peek() {
        if (isEmpty()) {
            return null;
        }
        return items.get(0);
    }

    public boolean isEmpty() {
        return items.size() == 0;
    }

    public int size() {
        return items.size();
    }

    public void clear() {
        items.clear();
    }

    public List<T> toArray() {
        return new ArrayList<>(items);
    }

    public void display() {
        System.out.println("Queue: " + items);
    }
} 

File: datastructures/python/queue.py
# Queue implementation in Python
# FIFO data structure with enqueue, dequeue, peek, isEmpty, size, clear, and toArray methods

from collections import deque

class Queue:
    def __init__(self):
        self.items = deque()
    
    def enqueue(self, element):
        self.items.append(element)
    
    def dequeue(self):
        if self.is_empty():
            return None
        return self.items.popleft()
    
    def peek(self):
        if self.is_empty():
            return None
        return self.items[0]
    
    def is_empty(self):
        return len(self.items) == 0
    
    def size(self):
        return len(self.items)
    
    def clear(self):
        self.items.clear()
    
    def to_array(self):
        return list(self.items)
    
    def display(self):
        print(f"Queue: {list(self.items)}") 